### PR TITLE
Adding VA and CO ETL from mapping for environmental justice

### PR DESCRIPTION
**Pull request by emma-nechamkin**
#237 Preliminary work on adding CO, VA data

### CODE REVIEW

**Comment by emma-nechamkin** [comment]
Some quick checks on threshold
Screen Shot 2022-01-25 at 5 18 52 PM

08 = Colorado, 51 = Virginia

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:83]: We could create a backlog ticket for that (although I'm not sure it's super high priority), or just email the authors of the data.

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:98]: Hmm, is `MAPPING_FOR_EJ_FINAL_RANK_FIELD` actually a percentile?

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:109]: Nit: in some of the older ETLs we defined `columns_to_keep` here in `load`, but we've recently moved to defining it as `self.COLUMNS_TO_KEEP` up in `__init__`. It can be helpful to keep the constant parameters up there so it's easy to understand the parameterizations at a glance.

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:235]: Similar comment as above, I don't think you need to reassign these but you can just reference `field_names.MAPPING_FOR_EJ_FINAL_RANK_FIELD` directly everywhere you use it. (And same goes for the other fields.)

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:242]: This is a nitty comment but I usually find acronyms and shortenings to be more trouble than they're worth -- so maybe move to just calling something like `mapping_for_ej_df`?

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:250]: Should we just make this a bool in the ETL class?

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:1005]: With all these black reformattings, I think (but I'm not 100% sure) there's a slight difference when you run `black .` from the command line on these files and when you run `black` inside of the notebook. My guess is that it's due to the additional quotes and linebreak characters causing the line limit to be exceeded when you look at the code from outside the notebook. 

Maybe we create a backlog ticket to investigate this and clean this up (e.g., exclude the notebooks from black when running from command line) so we don't keep shifting the code back and forth rather pointlessly.

File: data/data-pipeline/data_pipeline/score/field_names.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:451]: 🎉

**Review by lucasmbrown-usds — CHANGES_REQUESTED** [review]
I know this is a WIP but just figured I'd give you some feedback. Overall looking really good, nice work learning all about how to do this stuff in the codebase!

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:98]: yes! updated

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:250]: i think I am confused here -- this is a bool when it is loaded into a csv during the ETL, but I think it gets read as string from the CSV so we still have to specify the datatype. What am I missing? (an aside: I streamlined it a bit to specify the datatype on read; happy to change for calenviroscreen too if you think it's clearer this way?)

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:9]: ohhhhh right, my bad!

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:96]: got it. small nit: I usually interpret `PCT` as percent, so maybe we just spell out percentile in the variable name. or ignore this comment, it's not super high priority!

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb]: This is yet another undocumented convention, I'm not sure where the best place to write this down would be, but we've agreed to "clear output" from our python notebooks before committing changes to git to cleanup the git diff.

**Review by lucasmbrown-usds — COMMENTED** [review]
Looking good! Once you're up in S3, should be an easy approval.

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (esfoobar-usds) [reply data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:3702]: @emma-nechamkin it's easy to set "Format on Save" on VSCode and that would use Black. [Here's some instructions.](https://marcobelo.medium.com/setting-up-python-black-on-visual-studio-code-5318eba4cd00).

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:10]: Did you mean to resolve this? Would be great to clear these outputs before we merge. Thanks!

**Review by lucasmbrown-usds — COMMENTED** [review]
Looking good. Still waiting on S3, of course, but otherwise should be quick approval.

**Comment by emma-nechamkin** [comment]
One note here now that this largely works from AWS -- this data appears to be high-quality, etc (like, the percentiles are roughly flat and there are the exact right (1249, CO) or mostly right (VA) number of Census tracts, so I did not do any cleaning, really. There's a "missing_va" column in both datasets that appears to just count `NaN`s; I didn't fill `NaN`s but did not retain this column.

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:17]: 🎉

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:56]: you can delete this comment now, i think!

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:66]: nittiest nit award goes to: I'm sure they're the same by I believe the more pandas-y way to do this is `.astype(str)` instead of `map(str)`.

**Review by lucasmbrown-usds — APPROVED** [review]
Looks great! Small nits, but all good.

File: data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py:80]: I added this, but I'm wavering. The issue is that not every single tract has complete data in this dataset, and as a consequence, not every tract has a final score or percentile. This isn't a huge deal -- it's a handful of tracts -- but I don't love labeling the tracts as "not prioritized" simply because they lacked complete data. Consequently, I dropped them. The other two options -- allow null tracts not to be prioritized or have the prioritization column be `NaN`, `True`, or `False` -- seemed less foolproof to me, but I don't have particularly strong feelings here. @lucasmbrown-usds

Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/mapping_for_ej/etl.py]: Eh, to be consistent with previous ETLs, this does not happen 👍

**Pull request merged**
